Support move-only types in IPC messages.in, and use it for StreamServerConnection::Handle.

mattwoodrow opened the pull request:
#### 74200f5b920190e9e8b38e95a41583447b7463cb
<pre>
Support move-only types in IPC messages.in, and use it for StreamServerConnection::Handle.
<a href="https://bugs.webkit.org/show_bug.cgi?id=257528">https://bugs.webkit.org/show_bug.cgi?id=257528</a>

Reviewed by Kimmo Kinnunen.

Adds basic support to messages.py for creating rvalue references for types that are move only, and using WTFMove.

Makes StreamServerConnection::Handle move-only, since we never want to copy these (and the underlying connection state depends on their lifetime).

* Source/WebKit/Platform/IPC/StreamServerConnection.h:
(IPC::StreamServerConnection::Handle::encode):
(IPC::StreamServerConnection::Handle::decode):
(IPC::StreamServerConnection::Handle::encode const): Deleted.
* Source/WebKit/Scripts/webkit/messages.py:
(types_that_must_be_moved):
(function_parameter_type):
(arguments_constructor_name):
(message_to_struct_declaration):
* Source/WebKit/Scripts/webkit/tests/TestWithCVPixelBufferMessages.h:
(Messages::TestWithCVPixelBuffer::SendCVPixelBuffer::arguments):
(Messages::TestWithCVPixelBuffer::ReceiveCVPixelBuffer::arguments):
(Messages::TestWithCVPixelBuffer::SendCVPixelBuffer::arguments const): Deleted.
(Messages::TestWithCVPixelBuffer::ReceiveCVPixelBuffer::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithIfMessageMessages.h:
(Messages::TestWithIfMessage::LoadURL::arguments):
(Messages::TestWithIfMessage::LoadURL::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithImageDataMessages.h:
(Messages::TestWithImageData::SendImageData::arguments):
(Messages::TestWithImageData::ReceiveImageData::arguments):
(Messages::TestWithImageData::SendImageData::arguments const): Deleted.
(Messages::TestWithImageData::ReceiveImageData::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiverMessages.h:
(Messages::TestWithLegacyReceiver::LoadURL::arguments):
(Messages::TestWithLegacyReceiver::LoadSomething::arguments):
(Messages::TestWithLegacyReceiver::TouchEvent::arguments):
(Messages::TestWithLegacyReceiver::AddEvent::arguments):
(Messages::TestWithLegacyReceiver::LoadSomethingElse::arguments):
(Messages::TestWithLegacyReceiver::DidReceivePolicyDecision::arguments):
(Messages::TestWithLegacyReceiver::Close::arguments):
(Messages::TestWithLegacyReceiver::PreferencesDidChange::arguments):
(Messages::TestWithLegacyReceiver::SendDoubleAndFloat::arguments):
(Messages::TestWithLegacyReceiver::SendInts::arguments):
(Messages::TestWithLegacyReceiver::CreatePlugin::arguments):
(Messages::TestWithLegacyReceiver::RunJavaScriptAlert::arguments):
(Messages::TestWithLegacyReceiver::GetPlugins::GetPlugins):
(Messages::TestWithLegacyReceiver::GetPlugins::arguments):
(Messages::TestWithLegacyReceiver::GetPluginProcessConnection::arguments):
(Messages::TestWithLegacyReceiver::TestMultipleAttributes::arguments):
(Messages::TestWithLegacyReceiver::TestParameterAttributes::arguments):
(Messages::TestWithLegacyReceiver::TemplateTest::arguments):
(Messages::TestWithLegacyReceiver::SetVideoLayerID::arguments):
(Messages::TestWithLegacyReceiver::DidCreateWebProcessConnection::arguments):
(Messages::TestWithLegacyReceiver::InterpretKeyEvent::arguments):
(Messages::TestWithLegacyReceiver::DeprecatedOperation::arguments):
(Messages::TestWithLegacyReceiver::ExperimentalOperation::arguments):
(Messages::TestWithLegacyReceiver::LoadURL::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::LoadSomething::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::TouchEvent::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::AddEvent::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::LoadSomethingElse::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::DidReceivePolicyDecision::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::Close::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::PreferencesDidChange::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::SendDoubleAndFloat::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::SendInts::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::CreatePlugin::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::RunJavaScriptAlert::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::GetPlugins::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::GetPluginProcessConnection::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::TestMultipleAttributes::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::TestParameterAttributes::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::TemplateTest::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::SetVideoLayerID::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::DidCreateWebProcessConnection::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::InterpretKeyEvent::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::DeprecatedOperation::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::ExperimentalOperation::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithSemaphoreMessages.h:
(Messages::TestWithSemaphore::SendSemaphore::arguments):
(Messages::TestWithSemaphore::ReceiveSemaphore::arguments):
(Messages::TestWithSemaphore::SendSemaphore::arguments const): Deleted.
(Messages::TestWithSemaphore::ReceiveSemaphore::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBatchedMessages.h:
(Messages::TestWithStreamBatched::SendString::arguments):
(Messages::TestWithStreamBatched::SendString::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBufferMessages.h:
(Messages::TestWithStreamBuffer::SendStreamBuffer::arguments):
(Messages::TestWithStreamBuffer::SendStreamBuffer::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithStreamMessages.h:
(Messages::TestWithStream::SendString::arguments):
(Messages::TestWithStream::SendStringAsync::arguments):
(Messages::TestWithStream::SendStringSync::arguments):
(Messages::TestWithStream::CallWithIdentifier::arguments):
(Messages::TestWithStream::SendMachSendRight::arguments):
(Messages::TestWithStream::ReceiveMachSendRight::arguments):
(Messages::TestWithStream::SendAndReceiveMachSendRight::arguments):
(Messages::TestWithStream::SendString::arguments const): Deleted.
(Messages::TestWithStream::SendStringAsync::arguments const): Deleted.
(Messages::TestWithStream::SendStringSync::arguments const): Deleted.
(Messages::TestWithStream::CallWithIdentifier::arguments const): Deleted.
(Messages::TestWithStream::SendMachSendRight::arguments const): Deleted.
(Messages::TestWithStream::ReceiveMachSendRight::arguments const): Deleted.
(Messages::TestWithStream::SendAndReceiveMachSendRight::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclassMessages.h:
(Messages::TestWithSuperclass::LoadURL::arguments):
(Messages::TestWithSuperclass::TestAsyncMessage::arguments):
(Messages::TestWithSuperclass::TestAsyncMessageWithNoArguments::arguments):
(Messages::TestWithSuperclass::TestAsyncMessageWithMultipleArguments::arguments):
(Messages::TestWithSuperclass::TestAsyncMessageWithConnection::arguments):
(Messages::TestWithSuperclass::TestSyncMessage::arguments):
(Messages::TestWithSuperclass::TestSynchronousMessage::TestSynchronousMessage):
(Messages::TestWithSuperclass::TestSynchronousMessage::arguments):
(Messages::TestWithSuperclass::LoadURL::arguments const): Deleted.
(Messages::TestWithSuperclass::TestAsyncMessage::arguments const): Deleted.
(Messages::TestWithSuperclass::TestAsyncMessageWithNoArguments::arguments const): Deleted.
(Messages::TestWithSuperclass::TestAsyncMessageWithMultipleArguments::arguments const): Deleted.
(Messages::TestWithSuperclass::TestAsyncMessageWithConnection::arguments const): Deleted.
(Messages::TestWithSuperclass::TestSyncMessage::arguments const): Deleted.
(Messages::TestWithSuperclass::TestSynchronousMessage::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithoutAttributesMessages.h:
(Messages::TestWithoutAttributes::LoadURL::arguments):
(Messages::TestWithoutAttributes::LoadSomething::arguments):
(Messages::TestWithoutAttributes::TouchEvent::arguments):
(Messages::TestWithoutAttributes::AddEvent::arguments):
(Messages::TestWithoutAttributes::LoadSomethingElse::arguments):
(Messages::TestWithoutAttributes::DidReceivePolicyDecision::arguments):
(Messages::TestWithoutAttributes::Close::arguments):
(Messages::TestWithoutAttributes::PreferencesDidChange::arguments):
(Messages::TestWithoutAttributes::SendDoubleAndFloat::arguments):
(Messages::TestWithoutAttributes::SendInts::arguments):
(Messages::TestWithoutAttributes::CreatePlugin::arguments):
(Messages::TestWithoutAttributes::RunJavaScriptAlert::arguments):
(Messages::TestWithoutAttributes::GetPlugins::GetPlugins):
(Messages::TestWithoutAttributes::GetPlugins::arguments):
(Messages::TestWithoutAttributes::GetPluginProcessConnection::arguments):
(Messages::TestWithoutAttributes::TestMultipleAttributes::arguments):
(Messages::TestWithoutAttributes::TestParameterAttributes::arguments):
(Messages::TestWithoutAttributes::TemplateTest::arguments):
(Messages::TestWithoutAttributes::SetVideoLayerID::arguments):
(Messages::TestWithoutAttributes::DidCreateWebProcessConnection::arguments):
(Messages::TestWithoutAttributes::InterpretKeyEvent::arguments):
(Messages::TestWithoutAttributes::DeprecatedOperation::arguments):
(Messages::TestWithoutAttributes::ExperimentalOperation::arguments):
(Messages::TestWithoutAttributes::LoadURL::arguments const): Deleted.
(Messages::TestWithoutAttributes::LoadSomething::arguments const): Deleted.
(Messages::TestWithoutAttributes::TouchEvent::arguments const): Deleted.
(Messages::TestWithoutAttributes::AddEvent::arguments const): Deleted.
(Messages::TestWithoutAttributes::LoadSomethingElse::arguments const): Deleted.
(Messages::TestWithoutAttributes::DidReceivePolicyDecision::arguments const): Deleted.
(Messages::TestWithoutAttributes::Close::arguments const): Deleted.
(Messages::TestWithoutAttributes::PreferencesDidChange::arguments const): Deleted.
(Messages::TestWithoutAttributes::SendDoubleAndFloat::arguments const): Deleted.
(Messages::TestWithoutAttributes::SendInts::arguments const): Deleted.
(Messages::TestWithoutAttributes::CreatePlugin::arguments const): Deleted.
(Messages::TestWithoutAttributes::RunJavaScriptAlert::arguments const): Deleted.
(Messages::TestWithoutAttributes::GetPlugins::arguments const): Deleted.
(Messages::TestWithoutAttributes::GetPluginProcessConnection::arguments const): Deleted.
(Messages::TestWithoutAttributes::TestMultipleAttributes::arguments const): Deleted.
(Messages::TestWithoutAttributes::TestParameterAttributes::arguments const): Deleted.
(Messages::TestWithoutAttributes::TemplateTest::arguments const): Deleted.
(Messages::TestWithoutAttributes::SetVideoLayerID::arguments const): Deleted.
(Messages::TestWithoutAttributes::DidCreateWebProcessConnection::arguments const): Deleted.
(Messages::TestWithoutAttributes::InterpretKeyEvent::arguments const): Deleted.
(Messages::TestWithoutAttributes::DeprecatedOperation::arguments const): Deleted.
(Messages::TestWithoutAttributes::ExperimentalOperation::arguments const): Deleted.
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
(WebKit::IPCTestingAPI::JSIPCConnectionHandle::encode):
(WebKit::IPCTestingAPI::JSIPCStreamServerConnectionHandle::encode):
(WebKit::IPCTestingAPI::JSIPCConnectionHandle::encode const): Deleted.
(WebKit::IPCTestingAPI::JSIPCStreamServerConnectionHandle::encode const): Deleted.

Canonical link: <a href="https://commits.webkit.org/264848@main">https://commits.webkit.org/264848@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f8f2aba4144f65de13836d00070cc6cb13a605a2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/8847 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9135 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/9353 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/10500 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/8827 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11122 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9102 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11693 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/8993 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/9990 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/7826 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/10658 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/8972 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7288 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/15586 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/7604 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/8396 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8239 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/11573 "6 api tests failed or timed out") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/8490 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/8724 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/7110 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/9014 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7980 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/2150 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2142 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12191 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/9249 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/8472 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2269 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple commented on Source/WebKit/Platform/IPC/StreamServerConnection.h:
```diff
@@ -154,18 +163,20 @@ void StreamServerConnection::sendAsyncReply(AsyncReplyID asyncReplyID, Arguments
-        return std::nullopt;
-    return Handle { WTFMove(*outOfStreamConnection), WTFMove(*buffer) };
+        return result;
+    result.emplace(Handle(WTFMove(*outOfStreamConnection), WTFMove(*buffer)));
```
I think if you instead `return std::optional<Handle> { WTFMove(*outOfStreamConnection), WTFMove(*buffer) };` you get the inplace constructor that works with move-only types?

kkinnunen-apple commented on Source/WebKit/Platform/IPC/StreamServerConnection.h:
```diff
@@ -154,18 +163,20 @@ void StreamServerConnection::sendAsyncReply(AsyncReplyID asyncReplyID, Arguments
+inline void StreamServerConnection::Handle::encode(Encoder& encoder)
 {
-    encoder << outOfStreamConnection << buffer;
+    encoder << WTFMove(outOfStreamConnection) << buffer;
```
This leaves the instance in inconsistent state behind the callers back.
So I think part of the work in this patch is to ensure rvalue-only encoder functions work.
E.g. fix until `void encode(Encoder&) &&;` works...

mattwoodrow replied:
Oh that's a good point! It does work already, just need to add the `&&` ref-qualifier.

kkinnunen-apple commented on Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiverMessages.h:
```diff
@@ -333,18 +333,18 @@ class GetPlugins {
     static constexpr auto callbackThread = WTF::CompletionHandlerCallThread::ConstructionThread;
     using ReplyArguments = std::tuple<Vector<WebCore::PluginInfo>>;
-    explicit GetPlugins(bool refresh)
+    explicit GetPlugins(const bool& refresh)
```
This doesn't look right. `bool` parameters should be  passed by value, as before.

kkinnunen-apple commented on Source/WebKit/Scripts/webkit/tests/TestWithCVPixelBufferMessages.h:
```diff
@@ -56,9 +56,9 @@ class SendCVPixelBuffer {
     }
 
-    const auto& arguments() const
+    auto&& arguments()
```
The point of these tests is to test the code you modify.
So you could add a thing that shows how your new changes behave.

kkinnunen-apple commented on Source/WebKit/Scripts/webkit/messages.py:
```diff
@@ -152,16 +157,31 @@ def function_parameter_type(type, kind):
 
 
+def arguments_constructor_name(type, name):
+    move_types = frozenset([
```
This list is somehow twice here.

kkinnunen-apple commented on Source/WebKit/Scripts/webkit/messages.py:
```diff
@@ -152,16 +157,31 @@ def function_parameter_type(type, kind):
     if type in builtin_types:
         return type
 
+    if type in types_that_must_be_moved():
```
I think the IPC layer shouldn't enforce the "must be moved" on the count of:
 - Feature would be useful for copyable types
 - Theoretical perf difference since the caller can move the object into the call frame and doesn't need own storage

The type itself should enforce whether it's ok to copy.

So it would be better to frame it as:
- "Support for rvalue types" (as opposed to "Support move only types")
- is_by_value_type(name, kind) (as opposed to builtin_types, kind.startswith and types_that_must_be_moved)
- use std::forward if is_by_value_type() matches

mattwoodrow replied:
Sorry, I'm not sure I'm following what the code change would be for this.

Do we want to actually hold the object by value somewhere? I though the current approach of just forwarding the rval-reference through to the `encode` implementation was going to have the lowest overhead.

kkinnunen-apple replied:
Rvalue ref is a reference, e.g. a pointer to object that lives in somewhere.
When we hold by value, the object doesn't have to neccessarily live anywhere. 
So case like
```
send(Msg { funcReturningAHandle() })
--> "visualized"
By value:
send(Msg { funcReturningAHandle() }) // The temporary "is moved to the call frame of the function invocation". Since it is consumed by the call, it may be a "no-op"
By reference:
auto h = funcReturningAHandle(); // The temporary has to live somewhere
send(Msg { WTFMove(h) });
```

The former can operate on the value. If a move happens, it's a conceptual "no-op".
The later operates on the pointer. If move happens, it's a write to the object pointed via the pointer (the ref).

mattwoodrow replied:
Right, I think I'm mostly with you :)

Does that still work when `Msg` isn't an aggregate type (has a user-declared constructor), so we can't use `{ }` initialisation (I don't think?)?

If the constructor takes the object by value, and then moves the ctor parameter into `m_arguments`, would `funcReturningAHandle()` still avoid the temporary living somewhere?

kkinnunen-apple reviewed (approved):
Since you have everything working, supporting by-value objects can be fixed in future commits.

mattwoodrow committed:
Support move-only types in IPC messages.in, and use it for StreamServerConnection::Handle.
https://bugs.webkit.org/show_bug.cgi?id=257528

Reviewed by Kimmo Kinnunen.

Adds basic support to messages.py for creating rvalue references for types that are move only, and using WTFMove.

Makes StreamServerConnection::Handle move-only, since we never want to copy these (and the underlying connection state depends on their lifetime).

* Source/WebKit/Platform/IPC/StreamServerConnection.h:
(IPC::StreamServerConnection::Handle::encode):
(IPC::StreamServerConnection::Handle::decode):
(IPC::StreamServerConnection::Handle::encode const): Deleted.
* Source/WebKit/Scripts/webkit/messages.py:
(types_that_must_be_moved):
(function_parameter_type):
(arguments_constructor_name):
(message_to_struct_declaration):
* Source/WebKit/Scripts/webkit/tests/TestWithCVPixelBufferMessages.h:
(Messages::TestWithCVPixelBuffer::SendCVPixelBuffer::arguments):
(Messages::TestWithCVPixelBuffer::ReceiveCVPixelBuffer::arguments):
(Messages::TestWithCVPixelBuffer::SendCVPixelBuffer::arguments const): Deleted.
(Messages::TestWithCVPixelBuffer::ReceiveCVPixelBuffer::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithIfMessageMessages.h:
(Messages::TestWithIfMessage::LoadURL::arguments):
(Messages::TestWithIfMessage::LoadURL::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithImageDataMessages.h:
(Messages::TestWithImageData::SendImageData::arguments):
(Messages::TestWithImageData::ReceiveImageData::arguments):
(Messages::TestWithImageData::SendImageData::arguments const): Deleted.
(Messages::TestWithImageData::ReceiveImageData::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiverMessages.h:
(Messages::TestWithLegacyReceiver::LoadURL::arguments):
(Messages::TestWithLegacyReceiver::LoadSomething::arguments):
(Messages::TestWithLegacyReceiver::TouchEvent::arguments):
(Messages::TestWithLegacyReceiver::AddEvent::arguments):
(Messages::TestWithLegacyReceiver::LoadSomethingElse::arguments):
(Messages::TestWithLegacyReceiver::DidReceivePolicyDecision::arguments):
(Messages::TestWithLegacyReceiver::Close::arguments):
(Messages::TestWithLegacyReceiver::PreferencesDidChange::arguments):
(Messages::TestWithLegacyReceiver::SendDoubleAndFloat::arguments):
(Messages::TestWithLegacyReceiver::SendInts::arguments):
(Messages::TestWithLegacyReceiver::CreatePlugin::arguments):
(Messages::TestWithLegacyReceiver::RunJavaScriptAlert::arguments):
(Messages::TestWithLegacyReceiver::GetPlugins::GetPlugins):
(Messages::TestWithLegacyReceiver::GetPlugins::arguments):
(Messages::TestWithLegacyReceiver::GetPluginProcessConnection::arguments):
(Messages::TestWithLegacyReceiver::TestMultipleAttributes::arguments):
(Messages::TestWithLegacyReceiver::TestParameterAttributes::arguments):
(Messages::TestWithLegacyReceiver::TemplateTest::arguments):
(Messages::TestWithLegacyReceiver::SetVideoLayerID::arguments):
(Messages::TestWithLegacyReceiver::DidCreateWebProcessConnection::arguments):
(Messages::TestWithLegacyReceiver::InterpretKeyEvent::arguments):
(Messages::TestWithLegacyReceiver::DeprecatedOperation::arguments):
(Messages::TestWithLegacyReceiver::ExperimentalOperation::arguments):
(Messages::TestWithLegacyReceiver::LoadURL::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::LoadSomething::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::TouchEvent::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::AddEvent::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::LoadSomethingElse::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::DidReceivePolicyDecision::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::Close::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::PreferencesDidChange::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::SendDoubleAndFloat::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::SendInts::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::CreatePlugin::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::RunJavaScriptAlert::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::GetPlugins::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::GetPluginProcessConnection::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::TestMultipleAttributes::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::TestParameterAttributes::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::TemplateTest::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::SetVideoLayerID::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::DidCreateWebProcessConnection::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::InterpretKeyEvent::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::DeprecatedOperation::arguments const): Deleted.
(Messages::TestWithLegacyReceiver::ExperimentalOperation::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithSemaphoreMessages.h:
(Messages::TestWithSemaphore::SendSemaphore::arguments):
(Messages::TestWithSemaphore::ReceiveSemaphore::arguments):
(Messages::TestWithSemaphore::SendSemaphore::arguments const): Deleted.
(Messages::TestWithSemaphore::ReceiveSemaphore::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBatchedMessages.h:
(Messages::TestWithStreamBatched::SendString::arguments):
(Messages::TestWithStreamBatched::SendString::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBufferMessages.h:
(Messages::TestWithStreamBuffer::SendStreamBuffer::arguments):
(Messages::TestWithStreamBuffer::SendStreamBuffer::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithStreamMessages.h:
(Messages::TestWithStream::SendString::arguments):
(Messages::TestWithStream::SendStringAsync::arguments):
(Messages::TestWithStream::SendStringSync::arguments):
(Messages::TestWithStream::CallWithIdentifier::arguments):
(Messages::TestWithStream::SendMachSendRight::arguments):
(Messages::TestWithStream::ReceiveMachSendRight::arguments):
(Messages::TestWithStream::SendAndReceiveMachSendRight::arguments):
(Messages::TestWithStream::SendString::arguments const): Deleted.
(Messages::TestWithStream::SendStringAsync::arguments const): Deleted.
(Messages::TestWithStream::SendStringSync::arguments const): Deleted.
(Messages::TestWithStream::CallWithIdentifier::arguments const): Deleted.
(Messages::TestWithStream::SendMachSendRight::arguments const): Deleted.
(Messages::TestWithStream::ReceiveMachSendRight::arguments const): Deleted.
(Messages::TestWithStream::SendAndReceiveMachSendRight::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclassMessages.h:
(Messages::TestWithSuperclass::LoadURL::arguments):
(Messages::TestWithSuperclass::TestAsyncMessage::arguments):
(Messages::TestWithSuperclass::TestAsyncMessageWithNoArguments::arguments):
(Messages::TestWithSuperclass::TestAsyncMessageWithMultipleArguments::arguments):
(Messages::TestWithSuperclass::TestAsyncMessageWithConnection::arguments):
(Messages::TestWithSuperclass::TestSyncMessage::arguments):
(Messages::TestWithSuperclass::TestSynchronousMessage::TestSynchronousMessage):
(Messages::TestWithSuperclass::TestSynchronousMessage::arguments):
(Messages::TestWithSuperclass::LoadURL::arguments const): Deleted.
(Messages::TestWithSuperclass::TestAsyncMessage::arguments const): Deleted.
(Messages::TestWithSuperclass::TestAsyncMessageWithNoArguments::arguments const): Deleted.
(Messages::TestWithSuperclass::TestAsyncMessageWithMultipleArguments::arguments const): Deleted.
(Messages::TestWithSuperclass::TestAsyncMessageWithConnection::arguments const): Deleted.
(Messages::TestWithSuperclass::TestSyncMessage::arguments const): Deleted.
(Messages::TestWithSuperclass::TestSynchronousMessage::arguments const): Deleted.
* Source/WebKit/Scripts/webkit/tests/TestWithoutAttributesMessages.h:
(Messages::TestWithoutAttributes::LoadURL::arguments):
(Messages::TestWithoutAttributes::LoadSomething::arguments):
(Messages::TestWithoutAttributes::TouchEvent::arguments):
(Messages::TestWithoutAttributes::AddEvent::arguments):
(Messages::TestWithoutAttributes::LoadSomethingElse::arguments):
(Messages::TestWithoutAttributes::DidReceivePolicyDecision::arguments):
(Messages::TestWithoutAttributes::Close::arguments):
(Messages::TestWithoutAttributes::PreferencesDidChange::arguments):
(Messages::TestWithoutAttributes::SendDoubleAndFloat::arguments):
(Messages::TestWithoutAttributes::SendInts::arguments):
(Messages::TestWithoutAttributes::CreatePlugin::arguments):
(Messages::TestWithoutAttributes::RunJavaScriptAlert::arguments):
(Messages::TestWithoutAttributes::GetPlugins::GetPlugins):
(Messages::TestWithoutAttributes::GetPlugins::arguments):
(Messages::TestWithoutAttributes::GetPluginProcessConnection::arguments):
(Messages::TestWithoutAttributes::TestMultipleAttributes::arguments):
(Messages::TestWithoutAttributes::TestParameterAttributes::arguments):
(Messages::TestWithoutAttributes::TemplateTest::arguments):
(Messages::TestWithoutAttributes::SetVideoLayerID::arguments):
(Messages::TestWithoutAttributes::DidCreateWebProcessConnection::arguments):
(Messages::TestWithoutAttributes::InterpretKeyEvent::arguments):
(Messages::TestWithoutAttributes::DeprecatedOperation::arguments):
(Messages::TestWithoutAttributes::ExperimentalOperation::arguments):
(Messages::TestWithoutAttributes::LoadURL::arguments const): Deleted.
(Messages::TestWithoutAttributes::LoadSomething::arguments const): Deleted.
(Messages::TestWithoutAttributes::TouchEvent::arguments const): Deleted.
(Messages::TestWithoutAttributes::AddEvent::arguments const): Deleted.
(Messages::TestWithoutAttributes::LoadSomethingElse::arguments const): Deleted.
(Messages::TestWithoutAttributes::DidReceivePolicyDecision::arguments const): Deleted.
(Messages::TestWithoutAttributes::Close::arguments const): Deleted.
(Messages::TestWithoutAttributes::PreferencesDidChange::arguments const): Deleted.
(Messages::TestWithoutAttributes::SendDoubleAndFloat::arguments const): Deleted.
(Messages::TestWithoutAttributes::SendInts::arguments const): Deleted.
(Messages::TestWithoutAttributes::CreatePlugin::arguments const): Deleted.
(Messages::TestWithoutAttributes::RunJavaScriptAlert::arguments const): Deleted.
(Messages::TestWithoutAttributes::GetPlugins::arguments const): Deleted.
(Messages::TestWithoutAttributes::GetPluginProcessConnection::arguments const): Deleted.
(Messages::TestWithoutAttributes::TestMultipleAttributes::arguments const): Deleted.
(Messages::TestWithoutAttributes::TestParameterAttributes::arguments const): Deleted.
(Messages::TestWithoutAttributes::TemplateTest::arguments const): Deleted.
(Messages::TestWithoutAttributes::SetVideoLayerID::arguments const): Deleted.
(Messages::TestWithoutAttributes::DidCreateWebProcessConnection::arguments const): Deleted.
(Messages::TestWithoutAttributes::InterpretKeyEvent::arguments const): Deleted.
(Messages::TestWithoutAttributes::DeprecatedOperation::arguments const): Deleted.
(Messages::TestWithoutAttributes::ExperimentalOperation::arguments const): Deleted.
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
(WebKit::IPCTestingAPI::JSIPCConnectionHandle::encode):
(WebKit::IPCTestingAPI::JSIPCStreamServerConnectionHandle::encode):
(WebKit::IPCTestingAPI::JSIPCConnectionHandle::encode const): Deleted.
(WebKit::IPCTestingAPI::JSIPCStreamServerConnectionHandle::encode const): Deleted.

Canonical link: https://commits.webkit.org/264848@main
